tests/unit_tests/pyanaconda_tests/core/test_live_user.py

```diff
@@ -43,10 +47,59 @@ def test_get_live_user(self, getpwnam_mock, conf_mock):
                                            "USER": "liveuser",
                                            "HOME": "/home/liveuser",
                                        })
-        getpwnam_mock.assert_called_once_with("liveuser")
-        getpwnam_mock.reset_mock()
+        getpwuid_mock.assert_called_once_with(1024)
+        getpwuid_mock.reset_mock()
 
+    @patch("pyanaconda.core.live_user.conf")
+    @patch("pyanaconda.core.live_user.getpwuid")
+    @patch.dict("pyanaconda.core.live_user.os.environ", {"PKEXEC_UID": "1024"})
+    def test_get_live_user_with_failed_getpwuid(self, getpwuid_mock, conf_mock):
+        # supposedly live but missing user
+        conf_mock.system.provides_liveuser = True
+        getpwuid_mock.side_effect = KeyError
+        assert get_live_user() is None
+        getpwuid_mock.assert_called_once_with(1024)
```

Comment:
Ah, now I understand why you thought this is verbose compared to pytest! You can actually collapse all of the "fail" tests into one by using `patch.dict` within a loop:

```python
    @patch("pyanaconda.core.live_user.conf")
    @patch("pyanaconda.core.live_user.getpwuid")
    def test_get_live_user_with_wrong_pkexec_env(self, getpwuid_mock, conf_mock):
        """Test getting live user with wrong environment values"""
        # supposedly live but missing user
        conf_mock.system.provides_liveuser = True
        
        # no value
        with patch.dict("pyanaconda.core.live_user.os.environ", dict()):
            assert get_live_user() is None
            getpwuid_mock.assert_not_called()
        
        # all the rest
        for value in ("", "0", "-1", "999999999999999999999999", "hello world"):
            with patch.dict("pyanaconda.core.live_user.os.environ", {"PKEXEC_ENV": value}):
                assert get_live_user() == None
                getpwuid_mock.assert_not_called()
```

Reply:
Yeah, your suggested solution is probably better one. I was balancing between cleanliness and correctness.

Having each of these tests as separate test seems to me like more correct approach - one test / one test-case. However, yeah it's pretty verbose.